lib/Dialect/FIRRTL/Transforms/BlackBoxReader.cpp

```diff
@@ -164,14 +172,14 @@ void BlackBoxReaderPass::runOnOperation() {
     std::string output;
     llvm::raw_string_ostream os(output);
     for (auto &file : emittedFiles) {
-      const auto &fileName = file.fileName.getValue();
+      const auto &fileName = file.first.getValue();
       // Exclude Verilog header files since we expect them to be included
       // explicitly by compiler directives in other source files.
       auto ext = llvm::sys::path::extension(fileName);
       bool exclude = (ext == ".h" || ext == ".vh" || ext == ".svh");
-      file.op->setAttr(
+      file.second->setAttr(
           "output_file",
```

Comment:
doesn't this still create a verbatim node for each name, but then only set the output_file attribute on the last one?  wouldn't it bet better to not create unnecessary verbatim nodes?  Maybe have a map from name -> Text block rather than name -> verbatimOp.

Reply:
The test with -mlir gives:
```
  sv.verbatim "module ExtInline(); endmodule\0A"
  sv.verbatim "module ExtInline(); endmodule\0A" {output_file = {directory = "magic", exclude_from_filelist = false, exclude_replicated_ops = true, name = "blackbox-inline.v"}}
```
Which is bad, especially with single file verilog output.

Reply:
Slightly cleaner than a map to text blocks would be a map to resolved (existing) files, then loadFile is only run on the final file for a given name and files which won't wind up being the final verbatim node aren't loaded.

Reply:
Yeah, I noticed the code was horribly broken last night, I should have mentioned it in the PR.  I have some code that I think is similar to what you're saying, but it runs `loadFile()` early to canonicalize the path names, which will internally skip loading the file if its been seen before.